nucypher/policy/models.py

```diff
@@ -313,28 +307,141 @@ def make_arrangements(self, network_middleware,
             raise self.MoreKFragsThanArrangements
 
 
-class TreasureMap(object):
-    def __init__(self, m, ursula_interface_ids=None):
-        self.m = m
-        self.ids = set(ursula_interface_ids or set())
+class TreasureMap:
```

Comment:
Not sure how much I like this being in Characters, but we need to have this discussion on how to potentially restructure this.

Reply:
Can you clarify what you mean here?

Reply:
I don't like the idea of `TreasureMap` being in `Characters`. Previously, we discussed restructuring `Characters` into another module or something. Maybe this can also be discussed re: where would a better place for `TreasureMap` be?

Reply:
Do you mean literally in characters.py?  Has there been a proposal to move it there?  I agree - it doesn't belong there.  :-)

Or are you making a different point?